File: 백준/Silver/10828. 스택/스택.py
import sys

input = sys.stdin.readline
n = int(input())
stack = []

for i in range(n):
  command = input().split()
  if command[0] == "push":
    stack.append(command[1])
  elif command[0] == "pop":
    if stack:
      top = stack.pop()
      print(top)
    else:
      print(-1)
  elif command[0] == "size":
    print(len(stack))
  elif command[0] == "empty":
    if stack:
      print(0)
    else: 
      print(1)
  elif command[0] == "top":
    if stack:
      top = stack.pop()
      print(top)
      stack.append(top)
    else:
      print(-1)

File: 백준/Silver/10828. 스택/스택.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');

let n = input[0];
let stack = [];

for (let i = 1; i <= n; i++) {
  command = input[i].split(' ');
  if (command[0] == 'push') stack.push(command[1]);
  if (command[0] == 'pop') {
    if (stack.length == 0) console.log(-1);
    else {
      c = stack.pop();
      console.log(c);
    }
  }
  if (command[0] == 'size') console.log(stack.length);
  if (command[0] == 'empty') {
    if (stack.length == 0) console.log(1);
    else console.log(0);
  }
  if (command[0] == 'top') {
    if (stack.length == 0) console.log(-1);
    else {
      c = stack.pop();
      console.log(c);
      stack.push(c);
    }
  }
}
